tests/iam/test_scps.py

```diff
@@ -0,0 +1,262 @@
+    # To avoid collisions with earlier tests that my have been partially run
+    # Delete any roles by the same name that already exist
+    except botocore.exceptions.ClientError as error:
+        if error.response["Error"]["Code"] == "EntityAlreadyExists":
+            delete_role(role_name)
+            role = client.create_role(
+                RoleName=role_name,
+                AssumeRolePolicyDocument=json.dumps(assume_role_policy, indent=2),
+        )['Role']
+        else:
+            raise error
+    time.sleep(10)
+    yield Role(
+        role_name=role_name,
+        arn=role['Arn'],
+    )
+
+    reset_scps(TARGET_ACCOUNT_ID)
+    delete_role(role_name)
+
+
+# Note the importance of passing limited_role first. Must assume a role in the target account
+# before creating the test role so that the latter is also in the target account
+def test_implicit_deny_on_target_account(limited_role):
+    """
+    Does an scp on the target account limit access. Yes it does!
+    """
+
+    client = child_superclient("iam")
+    # Identity policy that permits something generic
+    idp = {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Sid": "statement1",
+                "Effect": "Allow",
+                "Action": [
+                    "s3:CreateBucket",
+                    "s3:DeleteBucket",
+                    "s3:PutObject" # Implicitly denied by scp
+                ],
+                "Resource": "*",
+            }
+        ]
+    }
+    # Attach idp
+    client.put_role_policy(
+        RoleName=limited_role.role_name,
+        PolicyName="AllowCreateBucket",
+        PolicyDocument=json.dumps(idp, indent=2),
+    )
+
+    # Use parent client to limit by SCP and switch to limited role
+    client = parent_client("organizations")
+    scp = {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Sid": "statement1",
+                "Effect": "Allow",
+                "Action": [
+                    "s3:CreateBucket",
+                    "s3:DeleteBucket"
+                    # NOTE: No "s3:PutObject" 
+                ],
+                "Resource": "*",
+            }
+        ]
+    }
+
+    policy_name = "allow-put-and-delete-bucket-only"
+    scp_id = client.create_policy(
+        Content=json.dumps(scp, indent=2),
+        Description="Limits actions in this account by implicit deny",
+        Name=policy_name,
+        Type="SERVICE_CONTROL_POLICY"
+    )["Policy"]["PolicySummary"]["Id"]
+    
+    client.attach_policy(
+        PolicyId=scp_id,
+        TargetId=TARGET_ACCOUNT_ID
+    )
+    # Detach allow all policy 
+    client.detach_policy(
+        PolicyId=FULL_ACCESS_POLICY_ID,
+        TargetId=TARGET_ACCOUNT_ID
+    )
+    time.sleep(10)
+
+    client = assumed_role_client("s3", target_role_arn=limited_role.arn, trusted_client=parent_client("sts"))
+
+    bucket_name = "scp-test-bucket-limited-role"
+    client.create_bucket(
```

Comment:
You may want to refactor the bucket into a test fixture or a context manager, as this has the same issue as other things, if you stop the tests after the bucket is created, it'll fail next time on this step.